### PR TITLE
Fix the fuck for Python 3.12

### PR DESCRIPTION
**Pull request by mbridon**
The imp module has been long deprecated and has been removed entirely in Python 3.12.

This solution is what was suggested in the release notes: https://docs.python.org/3.12/whatsnew/3.12.html#imp

### CODE REVIEW

**Comment by radrow** [comment]
finally someone gave a fuck

**Comment by mbridon** [comment]
> finally someone gave a fuck

I'd say I even gave "the fuck" 🤪

**Comment by otakuCoderr** [comment]
oh wow cool thanks was about to check it out but yeah u got the job done :+1:

**Comment by Payne680** [comment]
Waiting for the pull request to be merged 😎

**Comment by krossekrabbe** [comment]
This bit me after upgrading to Fedora 39.

**Comment by otakuCoderr** [comment]
> This ~bit~ fucked me after upgrading to Fedora 39.

lmao same still hasn't been merged :confused:

**Comment by radrow** [comment]
Just FYI due to this bug my terminal is fucked up since the upgrade:

image

No, I am not going to comment the fuck out of my zshrc.

**Comment by LoZeno** [comment]
Adding my +1 to get this merged soon!

**Comment by mbridon** [comment]
I don't have Python 2.7 any more, so I'll be happy if you can help fix this issue so the fix I provided works on both 2.7 and 3.12 :grin:

**Comment by lucazenari** [comment]
As a workaround, while we wait for this PR to be merged, if anyone wants to install this fork the quickest way is:
```
pip uninstall thefuck
pip install https://github.com/mbridon/thefuck/archive/main.zip
```

**Comment by tanenbaum** [comment]
I've just learned that there is an imp compatibility package called `python3-zombie-imp` on most distros (e.g. https://packages.fedoraproject.org/pkgs/python-zombie-imp/python3-zombie-imp/) that patches around this.

**Comment by mbridon** [comment]
> I've just learned that there is an imp compatibility package called `python3-zombie-imp` on most distros (e.g. https://packages.fedoraproject.org/pkgs/python-zombie-imp/python3-zombie-imp/) that patches around this.

Sure, that's a potential crutch, but fixing the issue is better 😉

**Review by Sisyphe42 — APPROVED** [review]

**Comment by fbe555** [comment]
Bump.
I also ran into this issue after updating to ubuntu 24.04 (and by extension to python 3.12.3), and fixed it by installing from the fork. Seems like this is super duper ready to merge :)

**Comment by mbridon** [comment]
Well, I guess the original author @nvbn seems to have disappeared, so all it would take is for someone to fork the repo and merge this PR and then assume maintenance of the project.

I won't do it for health reasons, so if someone who commented on this PR wants to have a go at it then go ahead. 🤷 

And when @nvbn comes back eventually, assume joint maintenance all together.

Free Software sometimes loses maintainers and find new ones, that's how it sometimes evolves and continues.

**Comment by nomike** [comment]
I believe this pull request could be closed.
A fix for this is already in the master branch.

**Comment by mbridon** [comment]
Ah great, so the maintainer fixed the issue but didn't merge the PR which could have given me the credit...

Well, all that matters is that it's fixed. :shrug:

**Comment by afunworm** [comment]
> Ah great, so the maintainer fixed the issue but didn't merge the PR which could have given me the credit...
> 
> Well, all that matters is that it's fixed. 🤷

Hello.

I am running the same issue on Ubuntu 24.04. However, your archive on your repository is no longer there. Is there any work-around to install this, system-wide?

Thank you.

**Comment by swirle13** [comment]
> Ah great, so the maintainer fixed the issue but didn't merge the PR which could have given me the credit...
> 
> Well, all that matters is that it's fixed. 🤷

Sorry to necro, but it appears the fix was made 3 months before your PR was opened Jul 2023 vs Nov 2023: https://github.com/nvbn/thefuck/commit/0420442e778dd7bc53bdbdb50278eea2c207dc74

The diff looks pretty similar to your PR diff [`904693a` (#1415)](https://github.com/nvbn/thefuck/pull/1415/commits/904693a496c5371bb8eb2d57698b720770c52f14)

**Comment by rezaebrh** [comment]
if linux wont let you install thefuck using pip, just clone the repo and install it using setup script

**Pull request closed without merging**
